src/uiScreens.c

```diff
@@ -21,6 +21,60 @@ void ui_displayPathScreen(
 	);
 }
 
+// the given path typically corresponds to an account
+// if it contains anything more, we display just the whole path
+void ui_displayAccountScreen(
+        const char* screenHeader,
+        const bip44_path_t* path,
+        ui_callback_fn_t callback
+)
+{
+	ASSERT(strlen(screenHeader) > 0);
+	ASSERT(strlen(screenHeader) < BUFFER_SIZE_PARANOIA);
+
+	ASSERT(bip44_hasValidCardanoPrefix(path));
+	ASSERT(bip44_containsAccount(path));
+
+	char accountDescription[160];
+	explicit_bzero(accountDescription, SIZEOF(accountDescription));
+
+	if (bip44_hasReasonableAccount(path) && !bip44_containsChainType(path)) {
+		uint32_t account = bip44_getAccount(path);
+		if (bip44_hasByronPrefix(path)) {
+			snprintf(
+			        accountDescription, SIZEOF(accountDescription),
+			        "Byron account %u  ", account
+			);
+		} else if (bip44_hasShelleyPrefix(path)) {
+			snprintf(
+			        accountDescription, SIZEOF(accountDescription),
+			        "Cardano account %u  ", account
```

Comment:
I tested this with Adalite and it shows me "Cardano account 2147483644 m/1852'/1815'/0'" so it apparently does not handle the hardened index properly

Also, imho the "Cardano" word is stealing too much real estate on the screen and it does not provide much information as the user already willingly chose to open the "Cardano" app (moreover, it can be misleading - Byron is also "Cardano") I would format the message like this instead (ordered by relevance):

`Account #<number> (Shelley/Byron) <path>`